src/server/services/UserService.ts

```diff
@@ -109,19 +109,19 @@ export default class UserService {
    */
   async updateUser(user: TESCUser) {
     return this.UserModel
-      .findOneAndUpdate({_id: user._id}, user)
+      .findOneAndUpdate({ _id: user._id }, user)
       .exec();
   }
 
   /**
    * Get a user's application to a given event.
-   * @param event The event associated with the application.
    * @param account The account that created the application.
+   * @param event The event associated with the application.
    * @param publicOnly Select only the public fields to return.
    */
-  async getUserApplication(event: TESCEvent, account: TESCAccount, publicOnly: boolean = false) {
+  async getUserApplication(account: TESCAccount, event?: TESCEvent, publicOnly: boolean = false) {
     let query = this.UserModel
-      .findOne({account: account, event: event});
+      .find({ account: account, ...event && { event: event } });
```

Comment:
This is a syntactic sugar for conditionally adding event if it exists, otherwise it leaves it out.
It's the same as
```
if (event) {
    .find({account: account, event: event})
} else {
    .find({account: account})
}